exchanges/ftx/ftx_test.go

```diff
@@ -0,0 +1,707 @@
+package ftx
+
+import (
+	"log"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+// Please supply your own keys here to do authenticated endpoint testing
+const (
+	apiKey                  = ""
+	apiSecret               = ""
+	canManipulateRealOrders = false
+)
+
+var f FTX
+
+func TestMain(m *testing.M) {
+	f.SetDefaults()
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../../testdata/configtest.json", true)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	exchCfg, err := cfg.GetExchangeConfig("FTX")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	exchCfg.API.AuthenticatedSupport = true
+	exchCfg.API.AuthenticatedWebsocketSupport = true
+	exchCfg.API.Credentials.Key = apiKey
+	exchCfg.API.Credentials.Secret = apiSecret
+
+	err = f.Setup(exchCfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func areTestAPIKeysSet() bool {
+	return f.ValidateAPICredentials()
+}
+
+// Implement tests for API endpoints below
+
+func TestGetMarkets(t *testing.T) {
+	t.Parallel()
+	_, err := f.GetMarkets()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetMarket(t *testing.T) {
+	t.Parallel()
+	_, err := f.GetMarket("FTT/BTC")
```

Comment:
Can make this a const as it makes it easy to adjust in the future with all these tests referencing it